Library/Homebrew/dev-cmd/pr-pull.rb

```diff
@@ -447,10 +449,16 @@ def self.pr_pull
       ohai "Fetching #{tap} pull request ##{pr}"
       Dir.mktmpdir pr do |dir|
         cd dir do
-          original_commit = ENV["GITHUB_SHA"].presence || tap.path.git_head
+          current_branch_head = ENV["GITHUB_SHA"] || tap.git_head
+          original_commit = if args.no_cherry_pick?
+            # TODO: Handle the case where `merge-base` returns multiple commits.
+            Utils.safe_popen_read("git", "-C", tap.path, "merge-base", "origin/HEAD", current_branch_head).strip
```

Comment:
This is needed because if we used the original definition of `original_commit`, `#formulae_need_bottles?` below mistakenly returns `false`. Looking at c669f1d625cd9795af397bc5a9fb78a77f45ac18 might give more context here.